lib/pure/sugar.nim

```diff
@@ -198,3 +198,65 @@ macro dump*(x: typed): untyped =
   let r = quote do:
     debugEcho `s`, " = ", `x`
   return r
+
+# TODO: consider exporting this in macros.nim
+proc replaceNodes(ast: NimNode): NimNode =
+  # Replace NimIdent and NimSym by a fresh ident node
+  # see also https://github.com/nim-lang/Nim/pull/8531#issuecomment-410436458
+  proc inspect(node: NimNode): NimNode =
+    case node.kind:
+    of nnkIdent, nnkSym:
+      return ident($node)
+    of nnkEmpty, nnkLiterals:
+      return node
+    else:
+      var rTree = node.kind.newTree()
+      for child in node:
+        rTree.add inspect(child)
+      return rTree
+  result = inspect(ast)
+
+macro distinctBase*(T: typedesc, recursive: static[bool] = false): untyped =
```

Comment:
/cc @mratsim @Araq 

> I think it should always be recursive. If I want to skip the 'distinct' indirections, I want to skip all of them.

same rationale as having both `getTypeImpl` and `getImpl`, plus other constructs where we want a direct vs a recursive behavior.

also, recursive = false is a saner default IMO:
`distinctBase(T)` will give clean compile error in case T is not a distinct type (ie, catches bugs)
`distinctBase(T, recursive = true)` will work (and recurse if applies), analog to the difference between "mkdir" and "mkdir -p"

Reply:
well indeed, I don't see the point of "mkdir" vs "mkdir -p", just create the directory recursively and stop wasting my time. ;-)

Reply:
I've changed it to be always recursive for the sake of moving on with this PR (but comments like "stop wasting my time" are not helpful as they don't address  the points I mentioned: eg, some applications require the non-recursive `getImpl`  and cannot work with its recursive variant `getTypeImpl`)

Reply:
I'm sorry, but "stop wasting my time" referred to mkdir's design.